clang/lib/CIR/Dialect/IR/CIRDialect.cpp

```diff
@@ -1274,6 +1274,33 @@ FunctionType CallOp::getCalleeType() {
   return FunctionType::get(getContext(), getOperandTypes(), getResultTypes());
 }
 
+//===----------------------------------------------------------------------===//
+// UnaryOp
+//===----------------------------------------------------------------------===//
+
+LogicalResult UnaryOp::verify() {
+  switch (getKind()) {
+  case cir::UnaryOpKind::Inc:
+    LLVM_FALLTHROUGH;
+  case cir::UnaryOpKind::Dec: {
+    // TODO: Consider looking at the memory interface instead of LoadOp/StoreOp.
+    auto loadOp = getInput().getDefiningOp<cir::LoadOp>();
+    if (!loadOp)
+      return emitOpError() << "requires input to be defined by a memory load";
+
+    for (const auto user : getResult().getUsers()) {
+      auto storeOp = dyn_cast<cir::StoreOp>(user);
+      if (storeOp && storeOp.getAddr() == loadOp.getAddr())
+        return success();
+    }
+    return emitOpError() << "requires result to be used by a memory store "
+                            "to the same address as the input memory load";
```

Comment:
Can you add a test for the error tracking in `clang/test/CIR/IR/invalid.cir`?

Reply:
Added, could you please take another look?